### PR TITLE
feat: Docker in docker as on option for executor k8s

### PR DESCRIPTION
**Pull request by djerraballi**
added annotations: 
    `screwdriver.cd/dockerEnabled: true` (if not present or disabled, no additional docker in docker container will be created) 
    `screwdriver.cd/dockerCpu: LOW` (same values as cpu) 
    `screwdriver.cd/dockerRam: LOW` (same values as ram)

1. had to switch out tinytim for handlebars since tinytim on node doesn't support conditional templating. 

2. this will create a docker-in-docker container along side your build container which enable you to run in docker build / run from within the executor_k8s pod.

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

File: index.js [hunk 1]
```diff
@@ -28,6 +32,13 @@ const PREFERRED_WEIGHT = 100;
 const ANNOTATIONS_PATH = 'metadata.annotations';
 const CONTAINER_WAITING_REASON_PATH = 'status.containerStatuses.0.state.waiting.reason';
 
+const DOCKER_ANNOTATIONS = [
+    'screwdriver.cd/dockerenabled',
```
Comment (tkyi) [inline index.js]: we've been using camel case
https://docs.screwdriver.cd/user-guide/configuration/annotations

File: index.js [hunk 2]
```diff
@@ -28,6 +32,13 @@ const PREFERRED_WEIGHT = 100;
 const ANNOTATIONS_PATH = 'metadata.annotations';
 const CONTAINER_WAITING_REASON_PATH = 'status.containerStatuses.0.state.waiting.reason';
 
+const DOCKER_ANNOTATIONS = [
+    'screwdriver.cd/dockerEnabled',
+    'screwdriver.cd/dockerCpu',
+    'screwdriver.cd/dockerRam'
```
Comment (minz1027) [inline index.js]: Thank you for contributing! this is going to be very helpful for a lot of ppl! 💃 
Can you put these annotations to the base? https://github.com/screwdriver-cd/executor-base/blob/master/index.js#L7
And you can remove the parseAnnotations function in here

Comment (djerraballi) [reply index.js]: https://github.com/screwdriver-cd/executor-base/pull/56/files

Comment (djerraballi) [reply index.js]: will have to merge that first and change the dep version in this pr

**Comment by djerraballi** [comment]
Updated with a dockerFeatureEnabled flag which will also override the dockerEnabled annotation, so the cluster admin will have to explicitly allow docker in order for the screwdriver annotation to work.

**Comment by minz1027** [comment]
sorry i merged the `queue-worker` one too fast! may need to open another one to pull these new dependencies once this one is merged

**Pull request merged**